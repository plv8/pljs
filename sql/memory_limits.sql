do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024/2);
$$ language pljs;

do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024/2);
$$ language pljs;

do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024);
$$ language pljs;

do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024/1.5);
  const s = [];
  while(true) {
    s.push(new ArrayBuffer(63)) // small non-aligned allocations
  }
$$ language pljs;
