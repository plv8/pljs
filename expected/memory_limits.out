do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024/2);
$$ language pljs;
do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024/2);
$$ language pljs;
do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024);
$$ language pljs;
ERROR:  execution error
DETAIL:  InternalError: out of memory
    at <anonymous> (<function>:4:28)
    at <eval> (<function>:6:3)

do $$
  const limit = pljs.execute(`select setting from pg_settings where name = $1`, ['pljs.memory_limit'])[0].setting;
  const a = new ArrayBuffer(limit*1024*1024/1.5);
  const s = [];
  while(true) {
    s.push(new ArrayBuffer(63)) // small non-aligned allocations
  }
$$ language pljs;
ERROR:  execution error
DETAIL:  out of memory
